tests/apps/demultiplex/test_sample_sheet_creator.py

```diff
@@ -136,3 +137,21 @@ def test_remove_unwanted_samples_no_dual_index(
         f"Removing sample {novaseq6000_flow_cell_sample_no_dual_index} since it does not have dual index"
         in caplog.text
     )
+
+
+def test_add_override_cycles_to_samples(
+    bcl_convert_sample_sheet_creator: SampleSheetCreatorBCLConvert,
+):
+    """."""
+    # GIVEN a SampleSheetCreator with samples without Override Cycles added
+    samples: List[FlowCellSampleBCLConvert] = bcl_convert_sample_sheet_creator.lims_samples
+
+    assert all(sample.override_cycles == "" for sample in samples)
+
+    # WHEN adding the correct values of override samples
+    bcl_convert_sample_sheet_creator.process_samples_for_sample_sheet()
+    samples: List[FlowCellSampleBCLConvert] = bcl_convert_sample_sheet_creator.lims_samples
+
+    # THEN the Override Cycles attribute is added to all samples
+    pattern: str = r"^Y.*I.*I.*Y.*$"
+    assert all(re.match(pattern, sample.override_cycles) for sample in samples)
```

Comment:
I would ideally like to see a test that checks if the correct cycles are added. This would also catch the potential bug that @Vince-janv have spotted. Might take some effort with the setup though.

Reply:
You are totally right. And this has to do with being more strict with tests. I will give it a try

Reply:
Take a look now @islean